src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,139 @@
+package duke.task;
+
+import duke.Message;
+import duke.IoManager;
+import duke.task.Task.Types;
+import duke.exception.ListEmptyException;
+import duke.exception.NoDescriptionException;
+import duke.exception.WrongNumberOfArgumentsException;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.regex.PatternSyntaxException;
+
+public class TaskManager {
+    public static final int MAX_TASKS = 100;
+
+    private static ArrayList<Task> tasks = new ArrayList<Task>(MAX_TASKS);
+
+    private interface Lambda {
+        void execute();
+    }
+    //TODO What do I even call this class?
+    private static class Runnable {
```

Comment:
Hmm these names are not really clear. May be `runTasks` for runnable? Also this piece of code looks a bit complicated (especially with lambdas). You might want to add some comments explaining the `WHAT` and `WHY` part of the code